Add GLM with design-based standard errors (#304)
xKDR/Survey.jl: docs/src/man/glm.md

Normal (IdentityLink)
Poisson (LogLink)
```

Refer to the GLM.jl documentation for more information about the GLM package.

## Fitting a GLM to a Survey Design object

You can fit a GLM to a Survey Design object the same way you would fit it to a regular data frame. The only difference is that you need to specify the survey design object as the second argument to the svyglm() function.

```julia
using Survey
apisrs = load_data("apisrs") 

# Simple random sample survey
srs = SurveyDesign(apisrs, weights = :pw) 

# Survey stratified by stype
dstrat = SurveyDesign(apistrat, strata = :stype, weights = :pw) 

# Survey clustered by dnum
dclus1 = SurveyDesign(apiclus1, clusters = :dnum, weights = :pw) 
```

Once you have the survey design object, you can fit a GLM using the svyglm() function. Specify the formula for the model and the distribution family. 

The svyglm() function supports all distribution families supported by GLM.jl, i.e. Bernoulli, Binomial, Gamma, Geometric, InverseGaussian, NegativeBinomial, Normal, and Poisson. 

For example, to fit a GLM with a Bernoulli distribution and a Logit link function to the `srs` survey design object we created above:

> ```julia

Reviewer: Can you make this a doctest?

formula = @formula(api00 ~ api99)

> my_glm = svyglm(formula, srs, family = Normal())

Reviewer: It's a bit confusing, since you have said “Bernoulli distribution and a Logit link function”, and then you are using a `Normal()` distribution family.  

The entire example should be for Bernoulli distribution.

# View the coefficients and standard errors
my_glm.Coefficients
my_glm.SE
```

## Examples

The examples below use the `api` datasets, which contain survey data collected about California schools. The datasets are included in the Survey.jl package and can be loaded by calling `load_data("name_of_dataset")`.

### Bernoulli with Logit Link

> Let us assume that the likelihood of receiving awards (`awards`) follows a Bernoulli distribution, which is characterized by modeling binary outcomes (0 or 1). Suppose we want to predict the likelihood of receiving awards based on the percentage of students eligible for subsidized meals (`meals`) and the percentage of English Language Learners (`ell`). We can fit this GLM using the code below:

Reviewer: ```suggestion
A school being eligible for the awards program (`awards`) is a binary outcome (0 or 1). Let's assume it follows a Bernoulli distribution. Suppose we want to predict `awards` based on the percentage of students eligible for subsidized meals (`meals`) and the percentage of English Language Learners (`ell`). We can fit this GLM using the code below:
```

1) Likelihood has a very specific meaning in statistics. 
2) `awards` indicates whether a school is Eligible for the awards program.

```julia
using Survey
apisrs = load_data("apisrs")
srs = SurveyDesign(apisrs, weights = :pw) 

# Convert yes/no to 1/0
apisrs.awards = ifelse.(apisrs.awards .== "Yes", 1, 0)

# Fit the model
model = glm(@formula(awards ~ meals + ell), apisrs, Bernoulli(), LogitLink())
```

### Binomial with Logit Link

> Let us assume that the number of students tested (`api_stu`) follows a Binomial distribution, which models the number of successes out of a fixed number of trials. Suppose we want to predict the number of students tested based on the percentage of students eligible for subsidized meals (`meals`) and the percentage of English Language Learners (`ell`). We can fit this GLM using the code below:

Reviewer: 1. 
```suggestion
Let us assume that the number of students tested (`api.stu`) follows a Binomial distribution, which models the number of successes out of a fixed number of trials. Suppose we want to predict the number of students tested based on the percentage of students eligible for subsidized meals (`meals`) and the percentage of English Language Learners (`ell`). We can fit this GLM using the code below:
```

1. Isn't Poisson to represent a count variable? @sourish-cmi @mousum-github 
2. @nadiaenh can you try to find out what the students are tested for? It'll make the paragraph a bit more complete.

Author: Yep correct, I think I was using another variable that was Binomial before and forgot to update that part. `api.stu` is the number of students in a given school that have undertaken a standardized test it seems
